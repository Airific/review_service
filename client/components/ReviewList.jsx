/* eslint-disable no-underscore-dangle */
/* eslint-disable react/prop-types */
import React from 'react';
import Styled from 'styled-components';
// eslint-disable-next-line import/extensions
import Review from './Review.jsx';

const ListGrid = Styled.div`
  display: flex;
  flex-flow: row wrap;

  justify-content: space-around;

`;

const Nested = Styled.div`
  width: 470px;
  margin-bottom: 20px;

`;

const ReviewList = ({ reviews = [] }) => {
  const list = reviews.slice(0, 6).map(
    (review) => (
      <Nested>
        <Review review={review} key={review._id} />
      </Nested>
    ),
  );
  return (
    <ListGrid>
      {list}
    </ListGrid>
  );
};

export default ReviewList;
