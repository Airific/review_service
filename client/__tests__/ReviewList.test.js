import React from 'react';
import { shallow } from 'enzyme';
import ReviewList from '../components/ReviewList';

const mockReviews = [
  {
    name: 'test',
    date: 'May',
    photo: 'https://airific-review-images.s3.us-east-2.amazonaws.com/image20.jpg',
    text: 'Hello!',
  },
];
describe('ReviewList component', () => {
  test('renders', () => {
    const wrapper = shallow(<ReviewList reviews={mockReviews} />);

    expect(wrapper).toMatchSnapshot();
  });
  test('return default empty array when there is no data to map through', () => {
    const wrapper = shallow(<ReviewList />);
    expect(wrapper).toMatchSnapshot();
  });
  test('does not break without reviews', () => {
    const wrapper = shallow(<ReviewList />);
    expect(wrapper.find('li')).toHaveLength(0);
  });
  test('does not break with an empty array', () => {
    const wrapper = shallow(<ReviewList reviews={[]} />);
    expect(wrapper.find('li')).toHaveLength(0);
  })
});
