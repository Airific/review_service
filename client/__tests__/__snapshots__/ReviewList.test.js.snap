// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList component renders 1`] = `
<styled.div>
  <styled.div>
    <Review
      review={
        Object {
          "date": "May",
          "name": "test",
          "photo": "https://airific-review-images.s3.us-east-2.amazonaws.com/image20.jpg",
          "text": "Hello!",
        }
      }
    />
  </styled.div>
</styled.div>
`;

exports[`ReviewList component return default empty array when there is no data to map through 1`] = `<styled.div />`;
